scheduler.go

```diff
@@ -264,6 +275,10 @@ func (s *Scheduler) runAndReschedule(job *Job) error {
 }
 
 func (s *Scheduler) run(job *Job) error {
+	if s.sem != nil {
+		s.sem.Acquire(context.Background(), 1)
```

Comment:
```suggestion
		s.sem.TryAcquire(context.Background(), 1)
		return nil
```

With that, in order to keep rescheduling working effectively, we'll need to bring this to after line 292, which isn't a problem anymore since we won't be locking